http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/SdkEventLoopGroup.java

```diff
@@ -105,6 +130,20 @@ public static SdkEventLoopGroup create(EventLoopGroup eventLoopGroup, ChannelFac
         return new SdkEventLoopGroup(eventLoopGroup, channelFactory);
     }
 
+    /**
+     * Creates a new instance of SdkEventLoopGroup with {@link EventLoopGroup}, {@link ChannelFactory}
+     * and {@link DatagramChannel} to be used with {@link NettyNioAsyncHttpClient}.
+     *
+     * @param eventLoopGroup the EventLoopGroup to be used
+     * @param channelFactory the channel factor to be used
+     * @param channelType the channel type to be used
+     * @return a new instance of SdkEventLoopGroup
+     */
+    public static SdkEventLoopGroup create(EventLoopGroup eventLoopGroup, ChannelFactory<? extends Channel> channelFactory,
+                                           Class<? extends DatagramChannel> channelType) {
+        return new SdkEventLoopGroup(eventLoopGroup, channelFactory, channelType);
+    }
```

Comment:
I think we can skip adding this constructor. If people really want to configure the datagram channel type, they can do it using the builder.

Reply:
I see, but then itwouldn't be posible to resolve channelType based on the eventLoopGroup, just like it's being done with the [channelFactory here](https://github.com/aws/aws-sdk-java-v2/blob/master/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/SdkEventLoopGroup.java#LL108C5-L120C6).

Reply:
I'm trying to avoid the pattern where we add a new constructor each time we add a new builder parameter. We generally only add a constructor if it's particularly common to create that combo. I don't think it's particularly common for people to use the `create(EventLoopGroup, ChannelFactory)` method we have today, and I'd expect even less common for people to use this new one with three parameters. If you think I'm wrong, I am fine adding it.

We could still allow for resolving the channelType based on the eventLoopGroup by keeping things private.

Reply:
You were right, I could resolve the datagramChannelFactory internally using the already existing constructors, so no additional constructor was added. 

I modified the builder to allow configuration of channel factories.

Changed pushed.